ntoskrnl/mm/section.c

```diff
@@ -1509,7 +1509,11 @@ MmAlterViewAttributes(PMMSUPPORT AddressSpace,
                  */
                 Page = MmGetPfnForProcess(Process, Address);
 
-                Protect = PAGE_READONLY;
+                /* Choose protection based on what was requested */
+                if (NewProtect == PAGE_EXECUTE_READWRITE)
+                    Protect = PAGE_EXECUTE_READ;
+                else
+                    Protect = PAGE_READONLY;
```

Comment:
Nitpick: Adding a empty line below or use `{}` would be better for readability.

Reply:
resolved.